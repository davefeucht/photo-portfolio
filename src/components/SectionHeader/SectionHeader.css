.section-header {
  background: var(--template-blue);
  padding: 5px;
  display: flex;
  flex-direction: row;
  align-items: center;
  color: var(--template-white);
}

.section-header > .title {
  font-size: 120%;
  font-weight: bold;
}

.section-header a {
  color: var(--template-white);
  text-decoration: none;
}