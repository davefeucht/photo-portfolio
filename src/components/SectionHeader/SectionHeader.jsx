/** ***************
* SectionHeader component implements the title bar of a Category or Page
**************** */

import './SectionHeader.css';

import PropTypes from 'prop-types';
import React from 'react';

function SectionHeader({ title }) {
    return (
        <div className="section-header">
            <div className="title">{title}</div>
        </div>
    );
}

SectionHeader.displayName = 'SectionHeader';

SectionHeader.propTypes = {
    title: PropTypes.string.isRequired
};

export default SectionHeader;
