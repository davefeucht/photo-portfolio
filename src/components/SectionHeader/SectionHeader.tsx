/** ***************
* SectionHeader component implements the title bar of a Category or Page
**************** */

import './SectionHeader.css';

import * as React from 'react';

interface SectionHeaderProps {
    title: string
}

const SectionHeader: React.FC<SectionHeaderProps> = ({ title }) => {
    return (
        <div className="section-header">
            <div className="title">{title}</div>
        </div>
    );
};

SectionHeader.displayName = 'SectionHeader';

export default SectionHeader;
