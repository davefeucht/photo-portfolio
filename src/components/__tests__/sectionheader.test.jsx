import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import SectionHeader from '../SectionHeader/SectionHeader';

const title = 'Some Section';

test('SectionHeader displays', () => {
    const component = renderer.create(
        <MemoryRouter>
            <SectionHeader title={title} />
        </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
