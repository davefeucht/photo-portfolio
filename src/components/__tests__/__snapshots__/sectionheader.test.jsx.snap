// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader displays 1`] = `
<div
  class="section-header"
>
  <div
    class="title"
  >
    Some Section
  </div>
</div>
`;
